src/main/java/org/folio/copycat/RecordRetriever.java

```diff
@@ -89,11 +89,28 @@ static void setAuthOptions(Connection conn, String auth) {
       ResultSet search = conn.search(query);
       Record record = search.getRecord(0);
       if (record == null) {
-        return Future.failedFuture("No record found");
+        return Future.failedFuture("No record found when searching "
+          + profile.getUrl() + " for identifier " + externalId);
       }
       return Future.succeededFuture(record.get(type));
     } catch (ZoomException e) {
-      return Future.failedFuture(e);
+      /*
+       https://github.com/indexdata/yaz4j/blob/master/src/main/java/org/yaz4j/ExceptionUtil.java
+       some messages include the server URL , some don't.. No sub classes for the exception
+       so we can determine if this is init or other error, and no way to get the error code.
+       Using this hack to determine if this was init error.
```

Comment:
Are the strings we're checking here from YAZ-ZOOM? Best be explicit about it in the comment. My first thought (probably wrong) was that they were from the OCLC server — which, if true, would mean that the heuristic would not work when running against the LoC server or others.

Also, I assume YAZ-ZOOM is not internationalized, so that if it _is_ the source of these messages, we can rely on them being in English? Again, best be explicit about this in the comment.

Reply:
Did you click on the link? Anyway these strings are hard-coded in yaz4j, so that would not depend on the target in any way.

Reply:
(ZOOM YAZ has diagnostic codes for everything, including non-bib1 things like connection lost, etc..)